cluster-autoscaler/config/autoscaling_options.go

```diff
@@ -169,4 +169,6 @@ type AutoscalingOptions struct {
 	DaemonSetEvictionForOccupiedNodes bool
 	// User agent to use for HTTP calls.
 	UserAgent string
+	// AzureEnableDynamicInstanceList enables dynamic instance workflow for vmss check
+	AzureEnableDynamicInstanceList bool
```

Comment:
usually not a fan of having provider-specific flags on the global flags. 

@gjtempleton thoughts on merging this with the AWS one? We could just have a generic `EnableDynamicInstanceInformation` flag and deprecate the AWS one eventually since I forsee other providers benefiting from that too.

Reply:
I'd be all in favour of merging it to a single flag, only challenge may be co-ordination with other cloud providers, I'm not sure what the current split is around default behaviours being dynamic or hard coded, and how many (if any) others support both.